Gems/AtomLyIntegration/CommonFeatures/Code/Source/Decals/EditorDecalComponent.cpp

```diff
@@ -97,13 +97,32 @@ namespace AZ
             BaseClass::Deactivate();
         }
 
-        AZ::Transform EditorDecalComponent::GetTransform() const
+        AZ::Transform EditorDecalComponent::GetWorldTransform() const
         {
             AZ::Transform transform = AZ::Transform::CreateIdentity();
             AZ::TransformBus::EventResult(transform, GetEntityId(), &AZ::TransformBus::Events::GetWorldTM);
             return transform;
         }
 
+        AZ::Vector3 EditorDecalComponent::GetNonUniformScale() const
+        {
+            AZ::Vector3 nonUniformScale;
+            AZ::NonUniformScaleRequestBus::EventResult(nonUniformScale, GetEntityId(), &AZ::NonUniformScaleRequests::GetScale);
```

Comment:
Is it possible the entity may not have a NonUniformScaleComponent? 
If so, you may want to initialize the nonUniformScale to (1, 1, 1)

Reply:
Thanks, you are totally right, not sure how I missed testing this.